Assets/MixedRealityToolkit/_Core/Interfaces/InputSystem/IMixedRealityInputSource.cs

```diff
@@ -1,32 +1,83 @@
     /// </summary>
     public interface IMixedRealityInputSource : IEqualityComparer
     {
+        /// <summary>
+        /// The ID assigned to the Input Source
+        /// </summary>
         uint SourceId { get; }
 
+        /// <summary>
+        /// The Name assigned to the Input Source
+        /// </summary>
         string SourceName { get; }
 
+        /// <summary>
+        /// Outputs the current state of the Input Source, whether it is tracked or not. As defined by the SDK / Unity.
+        /// </summary>
+        InputSourceState InputSourceState { get; }
+
+        /// <summary>
+        /// The designated hand that the Input Source is managing, as defined by the SDK / Unity.
+        /// </summary>
+        Handedness Handedness { get; }
+
+        /// <summary>
+        /// The list of Pointers attached to the Input Source
+        /// </summary>
         IMixedRealityPointer[] Pointers { get; }
 
         /// <summary>
-        /// Returns the input info that the input source can provide.
+        /// Details the list of available buttons / interactions available from the Input Source.
+        /// </summary>
+        Dictionary<InputType,InteractionDefinition> Interactions { get; }
+
+        /// <summary>
+        /// Returns the <see cref="Capabilities"/> that this Input Source can provide.
         /// </summary>
         InputType[] Capabilities { get; }
 
+        // TODO - Need to generalize the Input Type?
+        /// <summary>
+        /// Setup function to define controller
+        /// </summary>
+        /// <param name="state">SDK data to inform construction</param>
+        void SetupInputSource<T>(T state);
+
+        // TODO - Need to generalize the Input Type?
+        /// <summary>
+        /// Update function to update controller
+        /// </summary>
+        /// <param name="state">SDK data to inform update</param>
+        void UpdateInputSource<T>(T state);
+
         /// <summary>
-        /// Returns whether the input source supports the specified input types.
+        /// Returns whether the Input Source supports the specified input types.
         /// </summary>
         /// <param name="inputInfo">Input types that we want to get information about.</param>
-        bool SupportsInputCapability(InputType[] inputInfo);
+        bool SupportsCapabilities(InputType[] inputInfo);
+
+        /// <summary>
+        /// Returns whether the Input Source supports the specified input types.
+        /// </summary>
+        /// <param name="inputInfo">Input types that we want to get information about.</param>
+        bool SupportsCapability(InputType inputInfo);
+
+        /// <summary>
+        /// Register pointers against the controller
+        /// </summary>
+        /// <param name="pointers">Array of pointers to attach</param>
+        void RegisterPointers(IMixedRealityPointer[] pointers);
```

Comment:
That would be fair, but I leave to you to review since I'm unplugging inputsource from the controllers now.

Reply:
But the other Pointer code needs to go in to it's own static class for reuse, leaving only 

* RegisterPointers (meant to initialise or replace all current pointers)
* Unregister (although this could be achieved with RegisterPointers(null)

Possibly Register Pointers needs renaming if the array is intended to be modified through it.